src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -0,0 +1,208 @@
+    public class SearchCommand : Command
+    {
+        private readonly List<string> _sources = new List<string>();
+
+        [Option(typeof(NuGetCommand), "SearchCommandSourceDescription")]
+        public ICollection<string> Source
+        {
+            get { return _sources; }
+        }
+
+        // -Source "https://api.nuget.org/v3/index.json"
+
+
+        [Option(typeof(NuGetCommand), "SearchCommandPreRelease")]
+        public bool PreRelease { get; set; } = false;
+
+
+        [Option(typeof(NuGetCommand), "SearchCommandTake")]
+        public int Take { get; set; } = 20;
+
+
+        private IList<PackageSource> GetEndpointsAsync()
+        {
+            List<PackageSource> configurationSources = SourceProvider.LoadPackageSources()
+                .Where(p => p.IsEnabled)
+                .ToList();
+
+            IList<PackageSource> packageSources;
+            if (Source.Count > 0)
+            {
+                packageSources = Source
+                    .Select(s => PackageSourceProviderExtensions.ResolveSource(configurationSources, s))
+                    .ToList();
+            }
+            else
+            {
+                packageSources = configurationSources;
+            }
+            return packageSources;
+        }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            ILogger logger = Console;
+            CancellationToken cancellationToken = CancellationToken.None;
+
+            SearchFilter searchFilter = new SearchFilter(includePrerelease: PreRelease);
+
+            IList<PackageSource> listEndpoints = GetEndpointsAsync();
+
+            foreach (PackageSource source in listEndpoints)
+            {
+                string target = source.Source;
+                string name = source.Name;
+
+                SourceRepository repository = Repository.Factory.GetCoreV3(target);
+                PackageSearchResource resource = await repository.GetResourceAsync<PackageSearchResource>();
+
+                if (resource is null)
+                {
+                    continue;
+                }
+
+                IEnumerable<IPackageSearchMetadata> results = await resource.SearchAsync(
+                    string.Join("+", Arguments).Trim().Replace(' ', '+'), // The arguments (query strings) are joined with '+' so that the queries are passed to the
+                                                                          // search query url with the correct format (eg. "logging extensions" --> "logging+extensions")
+                    searchFilter,
+                    skip: 0,
+                    take: Take,
+                    logger,
+                    cancellationToken);
+
+                Console.WriteLine(new string('=', 20));
+                Console.WriteLine($"Source: {source.Name}");
+
+                if (!results.Any())
+                {
+                    Console.WriteLine(new string('-', 20));
```

Comment:
I'd extract this string to a local variable, since you're guaranteed to use it, so you can remove some redundant code.
Maybe even consider making it a class constant depending on how often it's used by the command.